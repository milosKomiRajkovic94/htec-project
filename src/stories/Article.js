import React from "react";

import PropTypes from "prop-types";

import "./article.css";

export const Article = ({ singleArticleData, onMore }) => {
  return (
    <div className="singleArticle">
      <div
        className={
          singleArticleData.title
            ? "singleArticleTitle"
            : "singleArticleTitle visibilityHidden"
        }
      >
        {singleArticleData.title}
      </div>
      <div className="singleArticleImage">
        <img
          src={
            singleArticleData.urlToImage
              ? singleArticleData.urlToImage
              : require("../assets/images/boxImage.png")
          }
          className="singleArticleImageTag"
          alt="singleArticleImageTag"
        />
      </div>
      <div
        className={
          singleArticleData.description
            ? "singleArticleDescription"
            : "singleArticleDescription visibilityHidden"
        }
      >
        {singleArticleData.description}
      </div>
      <div className="singleArticleMoreOption">
        <span
          className="cursorPointer"
          role="button"
          onClick={() => onMore()}
          onKeyPress={() => onMore()}
          tabIndex={0}
        >
          {"More >"}
        </span>
      </div>
    </div>
  );
};

Article.propTypes = {
  singleArticleData: PropTypes.shape({
    title: PropTypes.string,
    urlToImage: PropTypes.string,
    description: PropTypes.string,
  }),
  onMore: PropTypes.func,
};

Article.defaultProps = {
  singleArticleData: PropTypes.shape({
    title: "",
    urlToImage: "",
    description: "",
  }),
  onMore: () => {},
};
