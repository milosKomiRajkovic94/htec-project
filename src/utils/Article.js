import React from "react";

import PropTypes from "prop-types";

import { useTranslation } from "react-i18next";

const Article = ({ singleArticleData, onMore }) => {
  const { t } = useTranslation();

  return (
    <div className="singleArticle">
      <div
        className={
          singleArticleData.title
            ? "singleArticleTitle"
            : "singleArticleTitle visibilityHidden"
        }
      >
        {singleArticleData.title}
      </div>
      <div className="singleArticleImage">
        <img
          src={
            singleArticleData.urlToImage
              ? singleArticleData.urlToImage
              : require("../assets/images/boxImage.png")
          }
          className="singleArticleImageTag"
          alt="singleArticleImageTag"
        />
      </div>
      <div
        className={
          singleArticleData.description
            ? "singleArticleDescription"
            : "singleArticleDescription visibilityHidden"
        }
      >
        {singleArticleData.description}
      </div>
      <div className="singleArticleMoreOption">
        <span
          className="cursorPointer"
          role="button"
          onClick={() => onMore(singleArticleData)}
          onKeyPress={() => onMore(singleArticleData)}
          tabIndex={0}
          data-testid="more"
        >
          {`${t("more")} >`}
        </span>
      </div>
    </div>
  );
};

Article.propTypes = {
  singleArticleData: PropTypes.shape({
    title: PropTypes.string,
    urlToImage: PropTypes.string,
    description: PropTypes.string,
  }),
  onMore: PropTypes.func,
};

Article.defaultProps = {
  singleArticleData: PropTypes.shape({
    title: "",
    urlToImage: "",
    description: "",
  }),
  onMore: () => {},
};

export default Article;
