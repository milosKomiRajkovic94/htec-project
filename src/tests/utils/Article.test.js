import React from "react";

import ReactDOM from "react-dom";

import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Article from "../../utils/Article";

afterEach(cleanup);

it("renders without crashing with four required props", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Article
      singleArticleData={{
        title: "something",
        description: "something",
        url: "something",
      }}
    />,
    div
  );
});

it("renders language buttons correctly", () => {
  const { getByTestId } = render(
    <Article
      singleArticleData={{
        title: "something",
        description: "something",
        url: "something",
      }}
    />
  );
  expect(getByTestId("more")).toHaveTextContent("more");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Article
        singleArticleData={{
          title: "something",
          description: "something",
          url: "something",
        }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
