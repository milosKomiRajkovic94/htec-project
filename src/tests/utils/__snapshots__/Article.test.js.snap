// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="singleArticle"
>
  <div
    className="singleArticleTitle"
  >
    something
  </div>
  <div
    className="singleArticleImage"
  >
    <img
      alt="singleArticleImageTag"
      className="singleArticleImageTag"
      src="boxImage.png"
    />
  </div>
  <div
    className="singleArticleDescription"
  >
    something
  </div>
  <div
    className="singleArticleMoreOption"
  >
    <span
      className="cursorPointer"
      data-testid="more"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      more &gt;
    </span>
  </div>
</div>
`;
